src/_pytest/pathlib.py

```diff
@@ -629,6 +629,15 @@ def _import_module_using_spec(
 
     if spec_matches_module_path(spec, module_path):
         assert spec is not None
+
+        # Find spec and import this module.
```

Comment:
Thanks @dongfangtianyu for the PR.

> Based on this, I attempted to modify the execution process in _import_module_using_spec, moving the import of the parent module to after insert_missing_modules() to avoid a KeyError due to sys.modules not being ready.

But this change seems to be doing the opposite of what is recommended by https://docs.python.org/3/library/importlib.html#approximating-importlib-import-module: it is attempting to import the child module first, insert dummy/missing modules, and then import the parent module. The linked docs do the exact opposite.

Seems like inserting the dummy modules at this point might just be silencing the error with dummy modules instead of the correct modules.

Reply:
The original plan was to refactor this function by following the documentation example.

A shortcut was to use insert_missing_modules to simulate the keys in sys.modules, and unexpectedly, the tests passed.

I apologize for not tracking or analyzing the impact of this approach. I will confirm the implications as soon as possible.

Reply:
Confirmation: When loading test case files outside the cwd, the insert_missing_modules function will mock all levels of parent modules, which is not expected.